text/0000-compiler-requirements.md

```diff
@@ -0,0 +1,92 @@
+- [Designspace file format](https://fonttools.readthedocs.io/en/latest/designspaceLib/index.html): a serialized data format specified in XML
+- Glyphs file format: a serialized data format specified in ASCII plist format, with  fields that include Adobe OpenType Feature File Specification source code
+- [Monotype Source Data Format for OpenType Layout Tables](https://monotype.github.io/OpenType_Table_Source/otl_source.html): specifies typographic layout features
+- [Unified Font Object file format](https://unifiedfontobject.org/): a serialized data format specified in a combination of XML files, XML formatted property list  files, and Adobe OpenType Feature File Specification files
+
+### Current Tools
+
+fontmake. It's the glue that binds together ufo2ft, fontTools and some other codebases to take any of the usual font formats (UFOs, Designspace + UFOs and Glyphs.app files) and produce font binaries (TTF, CFF(2); static, interpolated static, variable).
+
+### Python Compiler Pipeline Libraries
+
+The Python library support requirements for font compilation include:
+
+- [fonttools](): I/O, lexing/parsing, intermediate code generation, optimization, code generation
+- [cu2qu](): intermediate code optimization
+- [glyphsLib](): I/O, lexing/parsing
+- [ufo2ft](): 
+- [MutatorMath](): linear interpolation calculation
+- [fontMath]()
+- [defcon]()
+- [booleanOperations]()
+- [ufoLib2]()
+- [attrs]()
+- [cffsubr]()
+- [compreffor]()
+- [ttfautohint-py]()
+
+### Object Code Specifications
+
+- otf
+- otf-cff2
+- ttf
+- ttf-interpolatable
+- otf-interpolatable
+- OpenType variable font ttf format
+- OpenType variable font cff2 format
+
+### Key Limitations
+
+- Python is bad at embarrasingly parallel tasks like instance generation and glyph compilation due to overhead and lacking multicore facilities.
+- The memory consumption of the compilation process can hinder the amount of build processes running in parallel.
+- It's easy to run into performance problems with OpenType Layout table repacking, which quickly consumes a large part of the compile time.
+- Codebases have grown organically by adding new things on top, some parts are byzantine and hard to change, making optimizations or rearranging code paths hard.
+  - VF generation requires compiling all masters fully and then merging them, instead of generating variable data directly
+  - fontmake is underloved from a maintenance perspective, with some low hanging performance fruits continuing to hang.
+  - The layering of fontmake → ufo2ft → fontTools makes changing the architecture hard. User-relevant options need to be made available and passed through all layers, e.g. GPOS compression recently
+  - Compilation of e.g. Glyphs.app files requires converting them into UFOs because ufo2ft can only compile those, which runs into impedance mismatches and is highly fragile and error-prone.
+
+## What Do We Need?
+
+An at least 10x faster compiler using less memory.
```

Comment:
It takes 30 minutes to compile Noto therefore 10x faster doesn't follow.

How often do people compile Noto? What happens if it takes half an hour? Why is there ROI for Google in making take half the time vs a tenth vs a hundredth, where does the return fall off?